### PR TITLE
Docs/fix vuepress build

### LINKED ISSUE #295: Vuepress documentation broken
**Describe the bug**
Vuepress docs build is broken

**To Reproduce**
`npm run docs:build`

**Expected behavior**
Build as usual

**Screenshots**

```json
Error: Rule can only have one resource source (provided resource and test + include + exclude) in {
  "exclude": [
    null
  ],
  "use": [
    {
      "loader": "cache-loader",
      "options": {
        "cacheDirectory": "/Users/jhgascon/Code/konviw/node_modules/@vuepress/core/node_modules/.cache/vuepress",
     ...
```

**Desktop (please complete the following information):**
- System:
  - OS: macOS 11.1
  - CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
  - Memory: 318.13 MB / 32.00 GB
  - Shell: 5.8 - /bin/zsh
- Binaries:
  - Node: 16.0.0 - ~/.nvm/versions/node/v16.0.0/bin/node
  - npm: 7.23.0 - ~/.nvm/versions/node/v16.0.0/bin/npm
- Browsers:
  - Chrome: 97.0.4692.71
  - Safari: 14.0.2

**Additional context**
I had recurrent issues with Vuepress in the past and too many extra dependencies that somehow pollute the konviw nestjs repository. 
An alternative would be the to move the docs to another repository or to move vuepress or vitepress into a folder with another package.json removing dependencies risks with konviw.

### PR DESCRIPTION
**Pull request by MeStrak**
Fixes #295

- Separate Vuepress package.json into docs folder to avoid dependency conflicts
- Add a ci for Vuepress build
  - Build on pull request so that it is now visible if a PR breaks the docs build
  - Auto publish gh-pages when main branch is updated

### CODE REVIEW

**Comment by PiiXiieeS** [comment]
I did not realise you managed this fix and great to see you could split to simplify the package.json for konviw.
In parallel I managed to upgrade to vuepress 2.0, see
https://github.com/Sanofi-IADC/konviw/pull/294/commits/f6eedd2ff3e3c67e3ab27e1e6f4b491cfaac774a
ready to merge.
We could merge that change and apply your changes to keep the idea to split the docs, which is better.

**Review by kiru42 — APPROVED** [review]
That's look good to me.

**Comment by PiiXiieeS** [comment]
I found some merge conflicts after merging https://github.com/Sanofi-IADC/konviw/pull/294 due to the migration to VuePress v2, so proposed a new commit with the fix.

File: docs/package.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "name": "konviw-docs",
+  "description": "Konviw documentation",
+  "scripts": {
+    "docs:prebuild": "rimraf .vuepress/dist && copyfiles ./CHANGELOG.md .",
+    "docs:dev": "vuepress dev .",
+    "docs:build": "vuepress build .",
+    "docs:deploy": "npm run docs:prebuild && npm run docs:build && gh-pages -d .vuepress/dist"
+  },
+  "license": "MIT",
+  "devDependencies": {
```
Comment (PiiXiieeS) [inline docs/package.json:11]: Moved the VuePress to v2.0.0-beta.33

**Review by PiiXiieeS — APPROVED** [review]
Moved VuePress to v2.0.0-beta.33 which as introduced in the previous merge request (https://github.com/Sanofi-IADC/konviw/pull/294)

**Pull request merged**